product/scanning.py

```diff
@@ -118,22 +124,41 @@ def upload_image(
 
         if scan_result:
             # check if image version changed
-
             metadata = scan_result.custom_data()
             image_reference = metadata.get('image-reference')
-            image_changed = image_reference == container_image.image_reference()
+            image_changed = image_reference != container_image.image_reference()
```

Comment:
could you split this (valid and good) refactoring change into a separate commit, please?